microprofile/config/src/main/java/io/helidon/microprofile/config/ConfigCdiExtension.java

```diff
@@ -164,475 +141,271 @@ private void harvestConfigPropertyInjectionPointsFromEnabledBean(@Observes Proce
-                                                                           typeArg2);
-                }
-            } else if (Map.class.isAssignableFrom(rawType)) {
-                return helidonConfig.get(configKey).detach().asMap();
-            } else if (Set.class.isAssignableFrom(rawType)) {
-                return new HashSet(helidonConfig.get(configKey).asList(typeArg)
-                                           .get());
-            } else {
-                throw new DeploymentException("Cannot create config property for " + rawType + "<" + typeArg + ">, key: "
-                                                      + configKey);
+                                                                    typeArg2);
             }
+        } else if (Map.class.isAssignableFrom(rawType)) {
+            return config.get(configKey).detach().asMap().get();
+        } else if (Set.class.isAssignableFrom(rawType)) {
+            return new LinkedHashSet<>(asList(config, configKey, typeArg, defaultValue));
+        } else {
+            throw new IllegalArgumentException("Cannot create config property for " + rawType + "<" + typeArg + ">, key: "
+                                                       + configKey);
         }
+    }
 
-        @Override
-        public Object create(CreationalContext<Object> context) {
-            Object value = getConfigValue(context);
-            if (null == value) {
-                throw MissingValueException.create(Config.Key.create(qualifier.key()));
-            }
-
-            return value;
-        }
+    private static <T> List<T> asList(Config config, String configKey, Class<T> typeArg, String defaultValue) {
+        try {
+            return config.get(configKey).asList(typeArg).get();
+        } catch (MissingValueException e) {
+            // if default
+            if (null == defaultValue) {
+                //noinspection ThrowInsideCatchBlockWhichIgnoresCaughtException
+                throw new NoSuchElementException("Missing list value for key "
+                                                         + configKey
+                                                         + ", original message: "
+                                                         + e.getMessage());
+            } else {
+                if (defaultValue.isEmpty()) {
+                    return Collections.emptyList();
+                }
 
-        private Object getConfigValue(CreationalContext<Object> context) {
-            return getConfigValue((Config) ConfigProvider.getConfig(), qualifier);
-        }
+                List<T> result = new LinkedList<>();
 
-        @Override
-        public Class<?> getBeanClass() {
-            return ConfigPropertyProducer.class;
-        }
-
-        @Override
-        public Set<InjectionPoint> getInjectionPoints() {
-            return Set.of();
-        }
+                String[] values = defaultValue.split(",");
```

Comment:
Maybe split on `"\\s*,\\s*`?

Reply:
The specification says it is comma separated. If I split on anything else, I may create more elements than desired